coderd/database/dbauthz/dbauthz.go

```diff
@@ -3677,6 +3677,20 @@ func (q *querier) UpdateUserLoginType(ctx context.Context, arg database.UpdateUs
 	return q.db.UpdateUserLoginType(ctx, arg)
 }
 
+func (q *querier) UpdateUserMustResetPassword(ctx context.Context, arg database.UpdateUserMustResetPasswordParams) error {
+	user, err := q.db.GetUserByID(ctx, arg.ID)
+	if err != nil {
+		return err
+	}
+
+	err = q.authorizeContext(ctx, policy.ActionUpdatePersonal, user)
```

Comment:
nit: this is fine for now, but I think it might make more sense to use a different action here. This would mean that users would be able to clear this themselves, whereas it's really the authentication subsystem itself doing it on behalf of the user.

Reply:
Ooh, I hadn't thought of that. I think that is a better idea. 

For in the future when setting the one-time-passcode we could then update `must_reset_password` based on if that is `NULL` or not? That would entirely avoid the need for this query.

Reply:
Yup!